genie-web/src/main/java/com/netflix/genie/web/services/impl/JobSpecificationServiceImpl.java

```diff
@@ -198,7 +201,13 @@ public JobSpecification resolveJobSpecification(final String id, @Valid final Jo
                 applicationResources,
                 this.generateEnvironmentVariables(id, jobRequest, cluster, command),
                 jobRequest.getRequestedAgentConfig().isInteractive(),
-                jobRequest.getRequestedAgentConfig().getRequestedJobDirectoryLocation().orElse(DEFAULT_JOB_DIRECTORY)
+                jobRequest.getRequestedAgentConfig().getRequestedJobDirectoryLocation().orElse(DEFAULT_JOB_DIRECTORY),
+                toArchiveLocation(
+                    jobRequest
+                        .getRequestedAgentConfig()
+                        .getRequestedArchiveLocationPrefix()
+                        .orElse(defaultArchiveLocation),
```

Comment:
Is the server always sending back an archive location if one was not specified?

Reply:
Yes, server always provides default location. This was a change after I had a discussion with Tom and had conveyed the same on #genie-developers

So there is no way to shut off archival.

Reply:
I missed the part where there's no way to opt-out. I don't think it's a good idea..